### PR TITLE
Alpha/mentor approval tables

### LINKED ISSUE #80: Mentor Approval Tables
**Description**
- Create tables necessary for mentor approval in database

**Requirements**
- [x] Create a table for that stores a mentor, an admin, and a boolean that says whether the admin has approved them
- [x] Add integer in MentorEvidence table for number of approvals
- [x] Add boolean in MentorEvidence for approval status
- [x] Add boolean in MentorEvidence for rejection status

### PR DESCRIPTION
**Pull request by OUMontiel**
# What does this PR do?
- Adds mentor approval tables

**Related Issues**
- Closes #80 

---
**Merge Checklist**
- [ ] Branch is up to date
- [ ] No code style issues

### CODE REVIEW

File: src/mysql/create.sql [hunk 1]
```diff
@@ -69,12 +69,28 @@ CREATE TABLE MentorExperience (
 -- Evidence of a user that is a mentor that proves them a previous intern.
 CREATE TABLE MentorEvidence (
   mentor_id INT NOT NULL,
+  approvals INT,
+  is_approved BOOLEAN,
+  is_rejected BOOLEAN,
```
Comment (shaargtz) [inline src/mysql/create.sql:74]: do we need is_rejected in evidence? wouldn't it be easier to just have the is_rejected in the Approval? what difference is there?

File: src/main/java/com/google/sps/classes/Notification.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+package com.google.sps.classes;
+
+import java.sql.Timestamp;
+
+/**
+ * Class to represent notifications and send them to front end as converted JSON.
+ */
+public class Notification {
+  public String message = "";
+  public String url = "";
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/classes/Notification.java:24]: Do you ever plan on using notifications without either of these fields initialized? If not, no need to initialize by default.

File: src/main/java/com/google/sps/classes/Utility.java [hunk 3]
```diff
@@ -16,7 +16,9 @@
 
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.classes.SqlConstants;
 import com.google.gson.Gson;
+import com.google.sps.classes.SqlConstants;
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/classes/Utility.java:21]: It looks like you're including this twice?

File: src/main/java/com/google/sps/servlets/NotificationServlet.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    // Get user's ID from Utility method.
+    int userId = Utility.getUserId();
+    // Fetch notifications if user is signed in and convert the ArrayList to JSON
+    // using Utility method.
+    response.setContentType("application/json;");
+    response.getWriter().println(Utility.convertToJsonUsingGson(getNotifications(userId)));
+  }
+
+  /**
+   * Posts a new notification for all followers of a modified question or answer.
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/servlets/NotificationServlet.java:52]: Is this your only notification type?

File: src/main/java/com/google/sps/servlets/NotificationServlet.java [hunk 5]
```diff
@@ -0,0 +1,195 @@
+// Copyright 2019 Google LLC
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/servlets/NotificationServlet.java:1]: All of the code in here seems unrelated to mentor approval tables, which is the title of this PR. Are these changes intended for this PR?

File: src/main/java/com/google/sps/servlets/SignupMentorServlet.java [hunk 6]
```diff
@@ -104,19 +111,50 @@ private void addMentorExperience(String[] experienceTags) {
...
       }
     }
   }
+
+  /**
+   * Inserts a new entry to MentorEvidence table with default values.
+   */
+  private void addMentorEvidence() {
+    int userId = Utility.getUserId();
+    String paragraph = "";
```
Comment (timothy-parsons) [inline src/main/java/com/google/sps/servlets/SignupMentorServlet.java:138]: Is this ever used?

File: src/main/java/com/google/sps/classes/Notification.java (hunk 2 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/classes/Notification.java]: This class was made by Andrés. Should I tell him this, add a TODO, change it myself...?

File: src/main/java/com/google/sps/classes/Utility.java (hunk 3 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/classes/Utility.java]: Whoops, removed!

File: src/main/java/com/google/sps/servlets/NotificationServlet.java (hunk 4 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/NotificationServlet.java]: As of this PR, it is. Again, this is Andrés's file. I made another PR where I add a new type of notification and I'm pretty sure I modify this :)

File: src/main/java/com/google/sps/servlets/NotificationServlet.java (hunk 5 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/NotificationServlet.java]: I had to pull Andrés's branch, but they're now in master, so this should not be reflected as a change.

File: src/main/java/com/google/sps/servlets/SignupMentorServlet.java (hunk 6 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/SignupMentorServlet.java:138]: Yes! Right at the end of the doPost() function.

**Comment by OUMontiel** [comment]
> This PR is HUGE. Think about how you can make your PRs smaller. For instance, there's a lot of code in here that doesn't seem related to 'mentor approval tables'. Also, your table logic should probably be in a PR separately from the servlet logic, which should be in a PR separately from hooking it into your javascript. I'm not going to ask you to change this PR (other than perhaps the notification logic that seems like it shouldn't be in here?), but please think about how you would split this PR up and try and practice that for future PRs.

I did try to keep it small. I mainly only changed the SQL files, but also added a bit to two servlets. I'll divide these changes into two PRs next time. Reason it was so huge was because I had to pull another branch, and those changes were being reflected here, but that should've not happened, and I'm sorry for this. If another of my PRs is this size, notify me and I'll make the changes necessary to make them smaller! :)

File: src/mysql/create.sql (hunk 1 above)
Comment (timothy-parsons) [reply src/mysql/create.sql:74]: Omar, just making sure you see this comment before I give the approval!

File: src/main/java/com/google/sps/servlets/SignupMentorServlet.java (hunk 6 above)
Comment (timothy-parsons) [reply src/main/java/com/google/sps/servlets/SignupMentorServlet.java:138]: I actually meant the 'paragraph' variable!

Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/SignupMentorServlet.java:138]: Oh, wow, it's not. Deleted!

File: src/mysql/create.sql (hunk 1 above)
Comment (OUMontiel) [reply src/mysql/create.sql:74]: Whoops, I did miss it, thanks for the bump! In another branch I do notice having is_approved and is_rejected in this table is unnecessary, so I change it to just is_reviewed.
